code/game/objects/items/weapons/grenades/frag.dm

```diff
@@ -71,3 +71,38 @@
 	blast_radius = 0
 	damage_step = 8
 	spread_range = 7
+
+/obj/item/grenade/frag/white_phosphorous
+	name = "SA WPG \"Geneva\""
+	desc = "A modernized Incendiary hailing popular use within technomancer tribes and Mercnary assualt teams. use this on unruly crowds."
+	icon_state = "white_pho"
+	item_state = "fraggrenade"
+	fragment_type = /obj/item/projectile/ember
+	var/datum/effect/effect/system/smoke_spread/white_phosphorous/smoke
+	num_fragments = 10
+	fragment_damage = 5
+	blast_radius = 0
+	damage_step = 5
+	spread_range = 7
+
+/obj/item/grenade/frag/white_phosphorous/prime()
+	playsound(loc, 'sound/effects/smoke.ogg', 50, 1, -3)
+	smoke.set_up(5, 0, usr.loc)
+	spawn(0)
+		smoke.start()
+		sleep(10)
+		smoke.start()
+		sleep(10)
+		smoke.start()
+		sleep(10)
+		smoke.start()
```

Comment:
```suggestion
	smoke.set_up(5, 0, get_turf(loc))
	smoke.start()
	..()
```
It produces a good amount of smoke and the fragments help spread it around. If it's too weak, you can always buff the smoke or increase the number of embers.